### ISSUE TITLE
Subscriber stats: All-time stats shows same values despite change in subscriber count over time

### ISSUE THREAD

**Issue by jp-imagines** [Automattic/wp-calypso#89066]
### Quick summary

Reported in 7930873-zd-a8c.

At _Stats > Subscribers_, the "all-time stats" section keeps track of the current total subscriber count, as well as the subscriber count from 30, 60, and 90 days ago. On this site, those counts are all the same despite the subscriber count changing over time.

### Steps to reproduce

1. Start with a site that is at least 90 days old and has had a change in total subscribers during the past few months.
2. Visit _Stats > Subscribers_.
3. Compare the four "all-time stats" counts (total, 30 days, 60 days, 90 days) with the values presented in the graph.

### What you expected to happen

Subscriber counts for 30, 60, and 90 days ago should match the subscriber point from those points in time, not the current subscriber count.

### What actually happened

Total subscriber count is reflected in all four data points.

Screenshot from user's Stats > Subscribers page:
![all time stats](https://github.com/Automattic/wp-calypso/assets/5630717/af554791-2eac-46d4-9bb3-d8faa9447917)

I see similar behavior for some internal sites as well, such as XkGC-p2:
![all time stats internal](https://github.com/Automattic/wp-calypso/assets/5630717/35e6e40b-af3b-48ce-af3e-a897e2809ccc)

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple, Atomic

**Comment by dzver** [comment]
I think it's supposed to show the number of subscribers 90 days ago, 60 days ago, 30 days ago, and now but now and 90 days ago are swapped or something else happened. This is a screenshot from my personal site which doesn't currently exhibit the issue - it still feels wrong and these numbers look random.

<img width="1433" alt="Screenshot 2024-04-02 at 11 42 42" src="https://github.com/Automattic/wp-calypso/assets/82778/cc1ba3dc-8dfd-42e4-8e27-d3975fc9c1fd">

**Comment by rickmgithub** [comment]
## 📌 REPRODUCTION RESULTS ##
- Tested on Atomic – Replicated

## 📌 FINDINGS/SCREENSHOTS/VIDEO ##
It does indeed show the same number. 

In my test case, 60 days ago the chart indicates the site had 35 subscribers. In the stats box for 60 days ago it shows the current total.

![Image](https://github.com/Automattic/wp-calypso/assets/13706336/6ffd0141-5133-48bd-a4fc-750fe2733c88)

## 📌 ACTIONS ##
Triaged

**Comment by andregardi** [comment]
I did some digging on Calypso side and concluded that the inconsistence is on API level.
Both the card and the chart consumes data from `/sites/${ siteId }/stats/subscribers` endpoint, but with different `quantity` param.

Each card makes one request for the specific date with `quantity: 1`.
The chart queries 30 days on a single request with `quantity: 30`.

Changing `quantity` results on different subscribers numbers for the same date, which is unexpected.
![Image](https://github.com/user-attachments/assets/4a02fd78-baa3-470d-a801-13891d3f2edc)

![Image](https://github.com/user-attachments/assets/197fe8e8-f515-4e2c-ad1f-3c13389eab3e)

**Comment by xavier-lc** [comment]
@kangzj I saw that you worked on the subscribers API, do you know what might be going on here? Thanks!

**Comment by kangzj** [comment]
I wonder whether this would be fixed by 177638-ghe-Automattic/wpcom as well @dognose24 ?

**Comment by annacmc** [comment]
> I wonder whether this would be fixed by 177638-ghe-Automattic/wpcom as well [@dognose24](https://github.com/dognose24) ?

I've given this a spin on a few different sites and so far can't repro, so perhaps it was fixed by that PR?

**Comment by dognose24** [comment]
> I wonder whether this would be fixed by 177638-ghe-Automattic/wpcom as well?

Yes, I think so, as we have now obtained accurate numbers from the changes. @andregardi, could you have a test for your case when available? 🙏🏼
